inference-engine/thirdparty/clDNN/src/impls/ocl/experimental_detectron_roi_feature_extractor.cpp

```diff
@@ -0,0 +1,88 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "cldnn/primitives/experimental_detectron_roi_feature_extractor.hpp"
+#include "experimental_detectron_roi_feature_extractor_inst.hpp"
+#include "primitive_base.hpp"
+#include "impls/implementation_map.hpp"
+#include "kernel_selector_helper.h"
+#include "experimental_detectron_roi_feature_extractor/experimental_detectron_roi_feature_extractor_kernel_selector.h"
+#include "experimental_detectron_roi_feature_extractor/experimental_detectron_roi_feature_extractor_kernel_ref.h"
+
+namespace cldnn {
+namespace ocl {
+struct experimental_detectron_roi_feature_extractor_impl : public typed_primitive_impl_ocl<experimental_detectron_roi_feature_extractor> {
+    using parent = typed_primitive_impl_ocl<experimental_detectron_roi_feature_extractor>;
+    using parent::parent;
+
+    std::unique_ptr<primitive_impl> clone() const override {
+        return make_unique<experimental_detectron_roi_feature_extractor_impl>(*this);
+    }
+
+protected:
+    kernel_arguments_data get_arguments(experimental_detectron_roi_feature_extractor_inst& instance, int32_t) const override {
+        kernel_arguments_data args;
+
+        for (std::size_t i = 0; i < instance.inputs_memory_count(); i++) {
+            args.inputs.push_back(instance.input_memory_ptr(i));
+        }
+        args.output = instance.output_memory_ptr();
+
+        return args;
+    }
+
+    event::ptr execute_impl(const std::vector<event::ptr>& events,
+                            experimental_detectron_roi_feature_extractor_inst& instance) override {
+        instance.copy_rois_input_to_second_output();
```

Comment:
This will physically copy full input buffer to second output buffer - is it expected?

Reply:
We should forward input to second output without changes

Reply:
I see
I think it is acceptable for initial implementation. But the better approach will be don't trigger physical memory copying and just reassign output pointer to it's input memory pointer